File: task/src/main/java/com/goexp/galgame/data/db/query/mongdb/CVQuery.java
package com.goexp.galgame.data.db.query.mongdb;

import com.goexp.common.db.mongo.DBQueryTemplate;
import com.goexp.common.db.mongo.ObjectCreator;
import com.goexp.galgame.common.model.CV;
import org.bson.Document;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class CVQuery {

    public static final DBQueryTemplate<CV> tlp = new DBQueryTemplate.Builder<>("galgame", "cv", new CVCreator()).build();

    private static class CVCreator implements ObjectCreator<CV> {

        private final Logger logger = LoggerFactory.getLogger(CVCreator.class);

        @Override
        public CV create(Document doc) {
            var g = new CV();

            logger.debug("<create> doc={}", doc);

            g.name = doc.getString("name");
            g.star = doc.getInteger("star");
            g.nameStr = doc.getString("names");
            return g;
        }
    }
}

File: task/src/main/scala/com/goexp/galgame/data/db/query/mongdb/CVQuery.scala
package com.goexp.galgame.data.db.query.mongdb

import com.goexp.common.db.mongo.{DBQueryTemplate, ObjectCreator}
import com.goexp.galgame.common.db.mongo.DB_NAME
import com.goexp.galgame.common.model.CV
import org.bson.Document
import org.slf4j.LoggerFactory

object CVQuery {
  lazy val tlp = new DBQueryTemplate.Builder[CV](DB_NAME, "cv", new CVCreator).build

  private class CVCreator extends ObjectCreator[CV] {
    private val logger = LoggerFactory.getLogger(classOf[CVCreator])

    override def create(doc: Document): CV = {
      logger.debug("<create> doc={}", doc)

      val g = new CV
      g.name = doc.getString("name")
      g.star = doc.getInteger("star")
      g.nameStr = doc.getString("names")
      g
    }
  }

}